### PR TITLE
MODORDERS-201 correct PO total estimated cost calculation

### PR DESCRIPTION
**Pull request by piotr-kalashuk**
## Purpose
[MODORDERS-201](https://issues.folio.org/browse/MODORDERS-201) Incorrect PO total estimated cost in some cases

## Approach
Replacing sum of regular `double` values by operation with `BigDecimal`

## Learning
[Why You Should Never Use Float and Double for Monetary Calculations](https://dzone.com/articles/never-use-float-and-double-for-monetary-calculatio)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/OrdersImplTest.java [hunk 1]
```diff
@@ -924,8 +930,10 @@ private void verifyCalculatedData(CompositePurchaseOrder resp) {
       .getCompositePoLines()
       .stream()
       .map(CompositePoLine::getCost)
-      .mapToDouble(Cost::getPoLineEstimatedPrice)
-      .sum();
+      .map(Cost::getPoLineEstimatedPrice)
```
Comment (KVupp) [inline src/test/java/org/folio/rest/impl/OrdersImplTest.java]: Just curious, any reason behind using the same code in unit tests too?
if we are using the exact same calculation that is used in the code for verification too,  will it be fair testing? the value will always end up the same , and isn't it partly the reason why we didn't know about the loss of precision?

Comment (craigmcnally) [reply src/test/java/org/folio/rest/impl/OrdersImplTest.java]: I agree that having a static expected value to compare against is probably better here.

**Comment by craigmcnally** [comment]
I know we ran into this same issue when first developing the mapper functionality, and I think I found the very same article.  I wonder if there's a better (move visible) place to document these sorts of pitfalls so that others on the team don't make the same mistake... especially since we're going to be doing a lot more monetary calculations once we start processing invoices and get into the finance side of things (payments/credits/transfers/etc.)  Any ideas?

File: src/test/java/org/folio/rest/impl/OrdersImplTest.java (hunk 1 above)
Comment (piotr-kalashuk) [reply src/test/java/org/folio/rest/impl/OrdersImplTest.java]: Moved to utility helper.

Comment (KVupp) [reply src/test/java/org/folio/rest/impl/OrdersImplTest.java]: The idea here is to use a assign a value vs the value being calculated even from a utility helper
for ex: double expectedPrice = 49.32 etc..

**Pull request merged**